### DISCUSSION TITLE
How to use Weights and Biases with AutoGluon?

### DISCUSSION THREAD

**Discussion by pj-mathematician** [autogluon/autogluon#2527] [Ideas]
I am using MultiModal predictor to classify images. I want to use [wandb](https://wandb.ai/) with it, so that I can use its features along with autogluon. Can you please guide me if this is possible without customizing the source code?

Since multimodal image prediction primarily uses PyTorch and PyTorch lightning, the example codes are as follows:
PyTorch:
```python
import wandb
# 1. Start a new run
wandb.init(project="gpt-3")
# 2. Save model inputs and hyperparameters
config = wandb.config
config.learning_rate = 0.01
# 3. Log gradients and model parameters
wandb.watch(model)
for batch_idx, (data, target) in enumerate(train_loader):
    if batch_idx % args.log_interval == 0:
    # 4. Log metrics to visualize performance
    wandb.log({"loss": loss})
```

PyTorch Lightning:
```python
from pytorch_lightning.loggers import WandbLogger
from pytorch_lightning import Trainer


wandb_logger = WandbLogger()
trainer = Trainer(logger=wandb_logger)
```

If this isnt currently possible then can you guide me a bit of the technicalities of the predictors and the frameworks they use so that I can fork and edit it accordingly 🙂 

Thanks

**Comment by vishal-kathuria** [comment]
Did you eventually figure this out? or just ended up customizing the source  code?

**Reply by pj-mathematician** [reply]
I settled for the tensorboard
